src/DependencyInjection/Compiler/ExtensionCompilerPass.php

```diff
@@ -153,15 +134,52 @@ protected function getExtensionsForAdmin($id, Definition $admin, ContainerBuilde
     /**
      * Resolves the class argument of the admin to an actual class (in case of %parameter%).
      *
-     * @return string
+     * @return string|null
+     *
+     * @phpstan-return class-string|null
      */
     protected function getManagedClass(Definition $admin, ContainerBuilder $container)
     {
-        return $container->getParameterBag()->resolveValue($admin->getArgument(1));
+        $argument = $admin->getArgument(1);
+        $class = $container->getParameterBag()->resolveValue($argument);
+
+        if (null === $class) {
+            // NEXT_MAJOR: Throw exception
+//            throw new \DomainException(sprintf('The admin "%s" does not have a valid manager.', $admin->getClass()));
+
+            @trigger_error(
+                sprintf('The admin "%s" does not have a valid manager.', $admin->getClass()),
+                E_USER_DEPRECATED
+            );
+        }
+
+        if (!\is_string($class)) {
+            // NEXT_MAJOR: Throw exception
+//            throw new \TypeError(sprintf(
+//                'Argument "%s" for admin class "%s" must be of type string, %s given.',
+//                $argument,
+//                $admin->getClass(),
+//                \is_object($class) ? \get_class($class) : \gettype($class)
+//            ));
+
+            @trigger_error(
+                sprintf(
+                    'Argument "%s" for admin class "%s" must be of type string, %s given.',
+                    $argument,
+                    $admin->getClass(),
+                    \is_object($class) ? \get_class($class) : \gettype($class)
+                ),
+                E_USER_DEPRECATED
+            );
+        }
+
+        return $class;
     }
 
     /**
-     * @return array an array with the following structure.
+     * @param array<string, array<string, array<string, string>>> $config
+     *
+     * @return array<string, array<string, array<string, array<string, array<string, string>>>>> an array with the following structure.
```

Comment:
```suggestion
     * @phpstan-return array<'excludes'|'admins'|'implements'|'extends'|'instanceof'|'uses', array<
     *                   string, array<string, array<string, array{priority, int}>>
     *               >>
     * @return array<string, array<string, array<string, array<string, array<string, int>>>>> an array with the 
```

or maybe both modulo some prefix?

Reply:
I'm not sure phpstan supports this syntax.

Reply:
Which part?

Reply:
`'excludes'|'admins'|'implements'|'extends'|'instanceof'|'uses'`

I was wrong https://phpstan.org/writing-php-code/phpdoc-types#literals-and-constants